src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1124,17 +974,329 @@
+
+    (sql/do-commands "INSERT INTO factsets_transform (id, certname, timestamp, environment_id, producer_timestamp)
+                      SELECT id, certname, timestamp, environment_id, timestamp
+                      FROM factsets fs")
+
+    (sql/create-table :fact_values_transform
+                      ["id" "bigint NOT NULL DEFAULT nextval('fact_values_id_seq')"]
+                      ["value_hash"    hash-type "NOT NULL"]
+                      ["value_type_id" "bigint NOT NULL"]
+                      ["value_integer" "bigint"]
+                      ["value_float"   "double precision"]
+                      ["value_string"  "text"]
+                      ["value_boolean" "boolean"]
+                      ["value_json"    "text"])
+
+    (sql/do-commands
+      (str "INSERT INTO fact_values_transform
+            (id, value_hash, value_type_id, value_integer, value_float, value_string, value_boolean, value_json)
+              SELECT id, " (munge-hash "value_hash") ", value_type_id, value_integer, value_float, value_string, value_boolean, value_json
+              FROM fact_values"))
+
+
+    (sql/create-table :resource_params_cache_transform
+                      ["resource" hash-type "NOT NULL"]
+                      ["parameters" "TEXT"])
+
+    (sql/do-commands
+      (str "INSERT INTO resource_params_cache_transform
+            (resource, parameters)
+              SELECT " (munge-hash "resource") ", parameters
+              FROM resource_params_cache"))
+
+    (sql/create-table :catalog_resources_transform
+                      ["catalog_id" "bigint NOT NULL"]
+                      ["resource"   hash-type "NOT NULL"]
+                      ["tags"       (sutils/sql-array-type-string "TEXT") "NOT NULL"]
+                      ["type"       "TEXT" "NOT NULL"]
+                      ["title"      "TEXT" "NOT NULL"]
+                      ["exported"   "BOOLEAN" "NOT NULL"]
+                      ["file" "TEXT"]
+                      ["line" "INT"])
+
+    (sql/do-commands
+      (str "INSERT INTO catalog_resources_transform
+            (resource, catalog_id, tags, type, title, exported, file, line)
+              SELECT " (munge-hash "resource") ", catalog_id, tags, type, title, exported, file, line
+              FROM catalog_resources"))
+
+    (sql/create-table :resource_params_transform
+                      ["resource" hash-type "NOT NULL"]
+                      ["name"  "TEXT" "NOT NULL"]
+                      ["value" "TEXT" "NOT NULL"])
+
+    (sql/do-commands
+      (str "INSERT INTO resource_params_transform
+            (resource, name, value)
+              SELECT " (munge-hash "resource") ", name, value
+              FROM resource_params"))
+
+    (sql/create-table :edges_transform
+                      ["certname" "TEXT" "NOT NULL"]
+                      ;; FIXME These should have FOREIGN KEY constraints on catalog_resources or resource_params_cache
```

Comment:
looks like you could put in an fk constraint on resource_params_cache(resource), but if it wasn't prexisting it's out of scope here I'd say. I was just pointing it out because you've marked it as something we need.  If we just remove the FIXME I'm happy.